src/components/usuarios/UsuarioForm.jsx

```diff
@@ -0,0 +1,268 @@
+import React, { Component } from 'react';
+import { firebaseConnect } from 'react-redux-firebase';
+import { withRouter } from 'react-router-dom';
+
+import axios from 'axios';
+
+import randomString from 'random-string';
+
+@firebaseConnect()
+@withRouter
+class UsuarioForm extends Component {
+
+  state = {
+    active: false,
+    role: '',
+    name: '',
+    email: '',
+    displayName: '',
+    password: '',
+    error: {
+      message: '',
+    },
+    isLoading: false,
+  }
+
+  constructor(props) {
+    super(props);
+
+    // If component recieves usuario as prop we merge it with initial state (used for editing)
+    this.state = { ...this.state, ...props.usuario };
+  }
+
+  addUsuario() {
+    const { active, role, name, email, displayName } = this.state;
+    let { password } = this.state;
+
+    // create context references for callback
+    const _this = this;
+    const { firebase } = this.props;
+
+    // generate random password if empty
+    if (password === '' || password === undefined) {
+      password = randomString({length: 10});
+    }
+
+    // create user function url
+    const createUser = 'https://us-central1-igv-andamiaje-co.cloudfunctions.net/createUser';
+
+    // loading. disables inputs
+    this.setState({ isLoading: true });
+
+    // call create user function
+    axios.get(createUser, {
+      params: {
+        email: email,
+        password: password,
+      },
+    	headers: {
+        'Access-Control-Allow-Origin': '*',
+    	},
+      mode: 'no-cors'
+    })
+    .then((response) => {
+      // use set() to push to user profile
```

Comment:
Is this overwriting the account/user just made on the server or write a profile on the server separate from the user account in firebase?

Reply:
this writes a different record on the database with the same ID as the user created in firebase auth